New package config for CellphoneDB

datasome opened the pull request:
_Name of the tool:_ CellphoneDB

_Short description:_ CellPhoneDB can be used to search for a particular ligand/receptor, or interrogate your own HUMAN single-cell transcriptomics data.

- The code is publicly available under an [OSI-approved](https://opensource.org/licenses/) license
     - Yes, [https://opensource.org/license/mit/](https://opensource.org/license/mit/)
 - The package provides versioned releases
     - Yes, see: [https://github.com/ventolab/CellphoneDB/tags](https://github.com/ventolab/CellphoneDB/tags)
 - The package can be installed from a standard registry (e.g. PyPI, conda-forge, bioconda)
    - Yes, see [https://pypi.org/project/CellphoneDB/](https://pypi.org/project/CellphoneDB/)
 - The package uses automated software tests and runs them via continuous integration (CI)
    - Yes, see: [https://github.com/ventolab/CellphoneDB/actions](https://github.com/ventolab/CellphoneDB/actions)
 - The package provides API documentation via a website or README
   - Yes, see: [cellphonedb.readthedocs.io](https://cellphonedb.readthedocs.io/en/latest/#) and https://www.cellphonedb.org/
 - The package uses scverse datastructures where appropriate (i.e. AnnData, MuData or SpatialData and their modality-specific extensions)
   - Yes, see: [cellphonedb.readthedocs.io](https://cellphonedb.readthedocs.io/en/latest/RESULTS-DOCUMENTATION.html#how-to-extract-the-cellphonedb-input-files-from-a-scanpy-anndata)
 - I am an author or maintainer of the tool and agree on listing the package on the scverse website
   - I am an author and a maintainer of the software and I agree on listing the package on the scverse website

grst commented on packages/CellphoneDB/meta.yaml:
```diff
@@ -0,0 +1,30 @@
+documentation_home: https://cellphonedb.readthedocs.io/en/latest/
+tutorials_home: https://github.com/ventolab/CellphoneDB/tree/master/notebooks
+publications:
+  - 10.1038/s41586-018-0698-6, 10.1038/s41596-020-0292-x, 10.1038/s41588-021-00972-2, 10.1038/s41586-022-04918-4
```
Please add publications as a yaml list, specifying one doi per line
```
 - 10.1038/s41586-018-0698-6
 - 10.1038/s41596-020-0292-x
```

datasome replied:
Done, thanks grst.

grst commented:
Hi @datasome, 

thanks for adding this (and sorry for the delay)! 
I will take a closer look once the tests pass. 

We are currently evaluating a partnership with [pyopensci](https://www.pyopensci.org/about-peer-review/index.html) for extended peer-review of ecosystem packages. This process will be more thorough, but also provide you with more valuable feedback regarding the structure and code quality of your package. 

If you are interested, I can suggest to perform a full package review for cellphonedb. If not, that's completely fine and I'll just evaluate according to our current checklist. 

Cheers, 
Gregor

datasome commented:
Hi Gregor,

Many thanks for replying and all your help with this. We would be satisfied with your current checklist for now if that’s ok. 
However, after CellphoneDB becomes a part of scverse ecosystem, it would be useful for us to get access to the full package review results also so that we can try and address any additional concerns as and when our resources allow - if that is possible.

Best regards,

Robert.

> On 22 Nov 2023, at 21:01, Gregor Sturm ***@***.***> wrote:
> 
> 
> Hi @datasome <https://github.com/datasome>,
> 
> thanks for adding this (and sorry for the delay)!
> I will take a closer look once the tests pass.
> 
> We are currently evaluating a partnership with pyopensci <https://www.pyopensci.org/about-peer-review/index.html> for extended peer-review of ecosystem packages. This process will be more thorough, but also provide you with more valuable feedback regarding the structure and code quality of your package.
> 
> If you are interested, I can suggest to perform a full package review for cellphonedb. If not, that's completely fine and I'll just evaluate according to our current checklist.
> 
> Cheers,
> Gregor
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/scverse/ecosystem-packages/pull/137#issuecomment-1823496983>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AJGK5SWK7CQU22B7ZAKSD6DYFZRZHAVCNFSM6AAAAAA7LBWPGWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTQMRTGQ4TMOJYGM>.
> You are receiving this because you were mentioned.
>

grst commented:
Hi @datasome, 

pretty much all boxes checked, I just have a question regarding the scverse data structures. Is it also possible to specify an in-memory AnnData object to the cellphonedb function? In the examples I saw in the documentation, there's always a path to an on-disk AnnData object specified, which seems a bit cumbersome for interactive analyses in an jupyter notebook. 

Other feedback (not related to the checklist): 

 * it's a convention to include a LICENSE file in the root of your repository with the full license text, e.g. https://github.com/scverse/cookiecutter-scverse-instance/blob/main/LICENSE
 * You are currently defining dependencies both in the pyproject.toml and `requirements.txt`. By using a modern build system such as `hatchling` (see the [PyPA packaging guide](https://packaging.python.org/en/latest/tutorials/packaging-projects/)), you can get rid of your `setup.py` and `requirements.txt` altogether. This simplifies things and cleans up redundancies. You can also take a look how it's done in the [scverse cookiecutter template](https://github.com/scverse/cookiecutter-scverse-instance). 

(a full pyopensci review would provide more of these sorts of feedback)

datasome commented:
Hi Gregor,

Many thanks for all your suggestions - please see my comments in turn:

- I've now added https://github.com/ventolab/CellphoneDB/blob/master/LICENSE
- I've removed setup.py and requirements.txt files as they were historical artefacts - we use [poetry](https://python-poetry.org/) to build the package prior to release
- Please see https://github.com/ventolab/CellphoneDB/commit/94268a8c20d81b38b81039c079b2a5a6d3fa9093 for a fix relating to your comment about allowing the user to pass an AnnData object instead of a file path to the counts file. 

Please let me know of any further comments and thanks again for your help on this.

Best,

Robert.

datasome commented:
Fab, many thanks for all your help Gregor, much appreciated!

Best,

Robert.
